src/main/kotlin/com/exactpro/th2/check1/CollectorService.kt

```diff
@@ -105,9 +106,17 @@ class CollectorService(
         val task = ruleFactory.createSequenceCheckRule(request)
 
         cleanupTasksOlderThan(olderThanDelta, olderThanTimeUnit)
+        val silenceCheck = if (request.hasSilenceCheck()) request.silenceCheck.value else defaultAutoSilenceCheck
+
+        val silenceCheckTask: AbstractCheckTask? = if (silenceCheck) {
+            ruleFactory.createSilenceCheck(request, olderThanTimeUnit.duration.toMillis() * olderThanDelta)
+        } else {
+            null
+        }
 
         eventIdToLastCheckTask.compute(CheckTaskKey(chainID, request.connectivityId)) { _, value ->
             task.apply { addToChainOrBegin(value, request.checkpoint) }
+                .run { silenceCheckTask?.also { subscribeNextTask(it) } ?: this }
```

Comment:
```suggestion
                .run {
                    val silenceCheck = if (request.hasSilenceCheck()) request.silenceCheck.value else defaultAutoSilenceCheck
                    if (silenceCheck) {
                        val silenceCheckTask = ruleFactory.createSilenceCheck(request, olderThanTimeUnit.duration.toMillis() * olderThanDelta)
                        subscribeNextTask(silenceCheckTask)
                        return@run silenceCheckTask
                    }
                    return@run this
                }
```
Maybe refactor like above

Reply:
I thought about this but the `compute` method should be as simple as possible. That's why I have decided to extract rule creation outside the `compute` method